alf/algorithms/algorithm.py

```diff
@@ -1050,11 +1081,28 @@ def update_with_gradient(self,
             assert len(loss_info.scalar_loss.shape) == 0
             loss_info = loss_info._replace(
                 loss=add_ignore_empty(loss_info.loss, loss_info.scalar_loss))
+        return loss_info
 
+    def _backward_and_gradient_update(
+            self,
+            loss_info,
+            weight=1.0,
+    ):
+        """Do backward and gradient update to all the trainable parameters.
+
+        Args:
+            loss_info (LossInfo): loss with shape :math:`(T, B)` (except for
```

Comment:
Seems no need to use LossInfo structure. Just a scalar loss should be fine.

Reply:
Correct. Changed to scalar loss input.